pkg/pillar/cmd/zedrouter/acl.go

```diff
@@ -254,162 +252,162 @@ func determineIpVer(isMgmt bool, bridgeIP string) int {
 	}
 }
 
-func applyACLRules(rules IptablesRuleList, bridgeName string, vifName string,
-	isMgmt bool, ipVer int, appIP string) error {
-
-	log.Debugf("applyACLRules: bridgeName %s ipVer %d appIP %s with %d rules\n",
-		bridgeName, ipVer, appIP, len(rules))
+func applyACLRules(aclArgs types.AppNetworkACLArgs,
+	rules types.IPTablesRuleList) (types.IPTablesRuleList, error) {
 	var err error
-	for _, rule := range rules {
-		log.Debugf("createACLConfiglet: rule %v\n", rule)
-		args := rulePrefix("-A", isMgmt, ipVer, vifName, appIP, rule)
-		if args == nil {
-			log.Debugf("createACLConfiglet: skipping rule %v\n",
-				rule)
+	var activeRules types.IPTablesRuleList
+	log.Debugf("applyACLRules: ipVer %d, bridgeName %s appIP %s with %d rules\n",
+		aclArgs.IPVer, aclArgs.BridgeName, aclArgs.AppIP, len(rules))
+	// the catch all drop rules are towards the end of the rule list
+	// hance we are inserting the rule in reverse order into
+	// the top of the target chain
+	numRules := len(rules)
+	for numRules > 0 {
```

Comment:
Please add comment about inserting in reverse order (to get in front of the drop rules for the network instance)

Reply:
sure, done. Also added checks for duplicate portmaps in itself, for an AppNetwork.